### PR TITLE
Use new privilege judge

### PR DESCRIPTION
**Pull request by jerry-yu**
Use new privilege pallet interface for  judge user's right

### CODE REVIEW

File: pallets/credit/src/lib.rs [hunk 1]
```diff
@@ -698,18 +697,23 @@ pub mod pallet {
             Self::do_unstaking_slash_credit(&user)
         }
 
-        #[pallet::weight(10_000 + T::DbWeight::get().reads_writes(0,1))]
-        pub fn set_credit_admin(origin: OriginFor<T>, admin: T::AccountId) -> DispatchResult {
-            ensure_root(origin)?;
-            CreditAdmin::<T>::put(admin.clone());
-            Self::deposit_event(Event::SetAdmin(admin));
+        #[pallet::weight(<T as pallet::Config>::WeightInfo::add_or_update_credit_data())]
+        pub fn external_set_credit_data(
```
Comment (xcaptain) [inline pallets/credit/src/lib.rs:701]: Not test covered. I think it's better to also add benchmark and weights, so in the future, just delete the `add_or_update_credit_data` related code

Comment (jerry-yu) [reply pallets/credit/src/lib.rs:701]: same logic as add_or_update_credit_data. when add_or_update_credit_data deprecated,  beachmark function name maybe changed

File: pallets/credit/src/lib.rs [hunk 2]
```diff
@@ -966,7 +970,19 @@ pub mod pallet {
...
+            let credit = credit_data.credit;
+            Self::do_add_credit(account_id.clone(), credit_data);
+            Self::deposit_event(Event::CreditUpdateSuccess(account_id, credit));
+        }
+
+        // using diff event for statistics
+        fn do_add_credit_with_other_event(account_id: T::AccountId, credit_data: CreditData) {
+            let credit = credit_data.credit;
+            Self::do_add_credit(account_id.clone(), credit_data);
+            Self::deposit_event(Event::CreditUpdated(account_id, credit));
```
[the remark this answers is not in the document]
Comment (jerry-yu) [reply pallets/credit/src/lib.rs:985]: new event added for  staking in bsc  when statistic stake info

[the remark this answers is not in the document]
Comment (jerry-yu) [reply pallets/credit/src/lib.rs:985]: I found the event StakingCreditScore declared but not used , Do use this event better ?

[the remark this answers is not in the document]
Comment (xcaptain) [reply pallets/credit/src/lib.rs:985]: Yes, I think it's good to keep `CreditUpdatedSuccess` and add a new event. `StakingCreditScore` is good, maybe give a better name if necessary.

**Review by xcaptain — APPROVED** [review]

**Pull request merged**